### ISSUE TITLE
NPM 6 - vulnerability - Memory Exposure

### ISSUE THREAD

**Issue by ghost** [request/tunnel-agent#41]
Moderate      │ Memory Exposure                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.6.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ node-sass                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ node-sass > request > tunnel-agent                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/598

**Comment by daniel-ac-martin** [comment]
Looks like this was fixed but the author neglected to publish v0.6.1 on NPM.

**Comment by thexpand** [comment]
When will v.0.6.1 be released on NPM?
https://www.npmjs.com/package/tunnel-agent

**Comment by igorescobar** [comment]
+1

**Comment by raitono** [comment]
I'm seeing the fix in my version I got from NPM. Just seems the version number wasn't updated to match. It was a simple one line change, so you should easily be able to see if you have the fix as well. Check line 131 of the index.js file. [Link to commit](https://github.com/request/tunnel-agent/commit/9ca95ec7219daface8a6fc2674000653de0922c0)

**Comment by mattjbrent** [comment]
its been almost a month. Can we please get the NPM package version updated please?

**Comment by ghost** [comment]
Please publish to NPM!?

**Comment by TomerADev** [comment]
@mikeal Please just run `npm publish`!

**Comment by Pablo-Araya** [comment]
+1

**Comment by ianmubangizi** [comment]
Am totally new to the npm audit, I want to know if I can actually continue developing without applying this fix because I have tried updating to the latest 0.6.0 tunnel-agent package but it won't fix the warning.

**Comment by ghost** [comment]
Please run npm publish

**Comment by igorescobar** [comment]
Please run npm publish

**Comment by jasonkhanlar** [comment]
Is this safe to use without the `npm publish` ?

**Comment by faelplg** [comment]
I am using `nvm` and since this issue is related to NPM +6, I fixed this with a downgrade to Node 10.2.1 (the last release with npm 5). See [the releases](https://nodejs.org/en/download/releases/).
Running `nvm install 10.2.1` has solved this for now.

**Comment by yetzt** [comment]
can someone press `npm publish` on this? @mikeal @simov

**Comment by yetzt** [comment]
ah, i just saw, its broken downstream; nevermind.

**Comment by joshwilkerson** [comment]
+1 ☝️

**Comment by ghost** [comment]
`npm run publish` 😭

**Comment by mikeal** [comment]
There are no commits in this repo since the last release so I'm not sure exactly what it is that I'm meant to be releasing?

**Comment by coughlanio** [comment]
@mikeal I think people are assuming there's a fix for the current NPM memory exposure audit error, why the version bump to 0.6.1 if there's no changes since 0.6.0?

```
 Moderate      │ Memory Exposure                                              │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ tunnel-agent                                                 │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=0.6.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ gatsby-plugin-sharp                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ gatsby-plugin-sharp > imagemin-webp > cwebp-bin >            │
│               │ bin-wrapper > download > caw > tunnel-agent                  │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/598   
```

**Comment by mikeal** [comment]
Odd release numbers don't get published, they are only in-tree, this makes it easier to see when people are using git checkouts rather than npm releases.

**Comment by sp90** [comment]
What is the status of this package?

**Comment by jasonkhanlar** [comment]
Oh my. I am still waiting on this to be fixed for npm also, re: https://github.com/request/tunnel-agent/issues/41?_pjax=%23js-repo-pjax-container#issuecomment-447755622
```
SEMVER WARNING: Recommended action is a potentially breaking change
```

**Comment by bakkerme** [comment]
For everyone coming to this issue after seeing this in their npm audit, the latest version, v0.6.0 has patched this issue. Most likely, the reason you are seeing the advisory is because a dependency nested somewhere in your application is relying on an unpatched version of this package, likely many layers down.

In the example above, `node-sass > request > tunnel-agent` is the dependency chain that linked to the vulnerable tunnel-agent. You can see this chain under the Path header in npm audit. It's possible that every part of this chain needs to update their dependencies to fix the vulnerability. 

Your best bet is to see if you're project's direct dependency, in this case `node-sass`, has updated their dependencies to a version that is fixed down the line.
